sonar-db/src/main/resources/org/sonar/db/component/ComponentMapper.xml

```diff
@@ -250,6 +250,40 @@
     ORDER BY UPPER(p.name), p.name
   </select>
 
+  <select id="selectByQuery" resultType="Component">
+    select
+    <include refid="componentColumns"/>
+    <include refid="sqlSelectByQuery" />
+    ORDER BY LOWER(p.name), p.name
```

Comment:
ok, this part will be improved when searching for components through a dedicated ES index.